tests/data/test_dataset.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane.data.Dataset` class.
+"""
+import pytest
+import numpy as np
+import pennylane as qml
+import dill
+import zstd
+from unittest.mock import patch, mock_open
+
+from pennylane.ops.qubit import hamiltonian
+
+
+def test_build_dataset():
```

Comment:
This looks okay for now, but let's try to port it to use `mock_open` working for this.